File: SnakeGameTS/src/models/syncdto.ts
export type SnakeSyncDto = {
    parts: { x: number, y: number }[]
}

export type FoodSyncDto = {
    x: number
    y: number
}

export type PlayerSyncDto = {
    id: number,
    snake: SnakeSyncDto,
    score: number,
    isGameOver: boolean
}

export type SyncDto = {
    players: PlayerSyncDto[],
    food: FoodSyncDto
}

File: SnakeGameTS/Models/SyncDto.cs
using SnakeGameTS.Objects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SnakeGameTS.Models
{
 
    public class SnakeSyncDto 
    {
        public PositionDto[] Parts { get; set; }
    }

    public class PlayerSyncDto {
        public long Id { get; set; }
        public SnakeSyncDto Snake { get; set; }
        public int Score { get; set; }
        public bool IsGameOver { get; set; }
    }

    public class SyncDto {
        public PlayerSyncDto[] Players { get; set; }
        public PositionDto Food { get; set; }
    }
}
